core/src/test/java/org/springframework/security/authentication/UserDetailsRepositoryReactiveAuthenticationManagerTests.java

```diff
@@ -140,4 +142,19 @@ public void authenticateWhenPasswordServiceAndUpgradeFalseThenNotUpdated() {
 
 		verifyZeroInteractions(this.userDetailsPasswordService);
 	}
+
+	@Test
+	public void authenticateWhenUserDetailsCheckerFails() {
+		when(this.userDetailsService.findByUsername(any())).thenReturn(Mono.just(this.user));
+		doThrow(new LockedException("account is locked")).when(this.userDetailsChecker).check(any());
+		when(this.encoder.matches(any(), any())).thenReturn(true);
+		this.manager.setPasswordEncoder(this.encoder);
+		this.manager.setPostAuthenticationChecks(this.userDetailsChecker);
+
+		assertThatExceptionOfType(LockedException.class)
+				.isThrownBy(() -> this.manager.authenticate(new UsernamePasswordAuthenticationToken(this.user, this.user.getPassword())).block())
+				.withMessage("account is locked");
+
+		verify(this.userDetailsChecker).check(eq(this.user));
+	}
 }
```

Comment:
Can you add another test that verifies that `postAuthenticationChecks` is not invoked if the `UserDetails` is not found?

Reply:
Done!